### ISSUE TITLE
Another issue for OnShutdown event

### ISSUE THREAD

**Issue by TheDeadCode** [StirlingLabs/MsQuic.Net#8]
I get this issue (but no pdb was found, tho):

```
System.InvalidCastException: 'Unable to cast object of type 'System.Delegate[]' to type 'StirlingLabs.Utilities.EventHandler`4[StirlingLabs.MsQuic.QuicPeerConnection,System.UInt64,System.Boolean,System.Boolean][]'.'
```

Sample code:
```csharp
connection.ConnectionShutdown += delegate(QuicPeerConnection sender, ulong arg1, bool b, bool arg3)
{
    Console.WriteLine("Connection is shutting down");
};
```

So sorry 'bout this gents.

**Comment by TYoungSL** [comment]
Did you have a stack trace?
Should have still gotten one without a pdb.

I'll add a test case to reproduce and fix what comes up.

If you can get me a stack trace I can make sure it covers the same code.

**Comment by TYoungSL** [comment]
Fixed in the last two commits.

Added tests to cover the affected code.

New release soon.

**Comment by TYoungSL** [comment]
@TheDeadCode We finally merged msquic v2.1.5 and we're preparing for a release.
